pygmt/src/solar.py

```diff
@@ -0,0 +1,88 @@
+"""
+solar - Plot day-night terminators.
+"""
+
+import datetime
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    R="region",
+    J="projection",
+    G="fill",
+    B="frame",
+    W="pen",
+    U="timestamp",
+    V="verbose",
+    X="xshift",
+    Y="yshift",
+    p="perspective",
+)
+@kwargs_to_strings(R="sequence", p="sequence")
+def solar(self, terminator="d", terminator_datetime="", **kwargs):
+    r"""
+    Plot day-light terminators.
+
+    This function plots the day-night terminator. Additionally, it can
+    plot the terminators for civil twilight, nautical twilight, and
+    astronomical twilight.
+
+    {aliases}
+
+    Parameters
+    ----------
+    terminator : str
+        Set the type of terminator displayed. The parameters are
+        ``day_night``, ``civil``, ``nautical``, and ``astronomical``, which
+        can be set with either the full name or the first letter of the name.
```

Comment:
```suggestion
        Set the type of terminator displayed. Valid arguments are
        **day_night**, **civil**, **nautical**, and **astronomical**, which
        can be set with either the full name or the first letter of the name.
```
Sorry for being part of the Parameters vs Arguments police :stuck_out_tongue_winking_eye: